src/Build/BackEnd/Components/SdkResolution/SdkResolverCachingWrapper.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Collections;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.BackEnd.SdkResolution
+{
+
+    internal sealed class SdkResolverCachingWrapper: ISdkResolverService
```

Comment:
I'm not in love with the need for a wrapper, is there any way to just turn caching on for a particular instance of SdkResolverService?

Reply:
True, in this particular case there is not much of the wrapper. My initial hope was to also remove the cache from the out of proc sdk service, but that one turns out be a different cache. As discussed, I'll push the cache into a subclass of SdkResolverService.